src/dbnode/storage/index.go

```diff
@@ -986,6 +1012,32 @@ func (i *nsIndex) query(
 	results index.BaseResults,
 	opts index.QueryOptions,
 	execBlockFn execBlockQueryFn,
+) (bool, error) {
+	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxSubQuery)
```

Comment:
Var naming nit: mind calling this NSIdxQueryHelper instead? SubQuery has a connotation best avoided here

Reply:
Yeah agreed - `subQuery` is weird..